src/utils/nextEntityId.ts

```diff
@@ -1,15 +1,40 @@
 import { EntityManager } from 'typeorm'
 import { NextEntityId } from '../model'
+import { AnyEntity, Constructor, EntityManagerOverlay } from './overlay'
 
-// used to retrieve the next id for an entity
-export async function getNextIdForEntity(em: EntityManager, entityName: string): Promise<number> {
+// used to retrieve the next id for an entity from NextEntityId table using either EntityManager or Overlay
+export async function getNextIdForEntity(
+  store: EntityManager | EntityManagerOverlay,
+  entityName: string
+): Promise<number> {
+  // Get next entity id from overlay (this will mostly be used in the mappings context)
+  if (store instanceof EntityManagerOverlay) {
+    const row = await store
+      .getRepository(NextEntityId as Constructor<NextEntityId & AnyEntity>)
+      .getOneBy({ entityName: entityName })
+
+    const id = parseInt(row?.nextId.toString() || '1')
+
+    // Update the id to be the next one in the overlay
+    if (row) {
```

Comment:
where did we increment the id for entity previously?

Reply:
In the [addRuntimeNotification](https://github.com/Joystream/orion/blob/master/src/utils/notification/helpers.ts#L219) function